### DISCUSSION TITLE
Is there anyway to tell if a type a global/built-in type

### DISCUSSION THREAD

**Discussion by awareness481** [dudykr/stc#986] [Q&A] [answered]
```ts
interface Array<T> {
    x: any;
}
```

For example here, is there any way to tell that `Array<T>` is not a new type but this is a declaration merge because `Array<T>` already exists from the builtin types?

**Comment by kdy1** [comment, marked as answer]
You can use [`env.get_global_type`](https://rustdoc.stc.dudy.dev/stc_ts_env/struct.Env.html#method.get_global_type).

Example: https://github.com/dudykr/stc/blob/32d80edb2c18fa55ed57cbd9ea46af35853d5052/crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs#L1072

**Reply by awareness481** [reply]
Thanks!

**Comment by sunrabbit123** [comment]
This seems to have been answered, so close it.
